crates/payout_curve/src/lib.rs

```diff
@@ -0,0 +1,685 @@
+use anyhow::Context;
+use anyhow::Result;
+use rust_decimal::prelude::ToPrimitive;
+use rust_decimal::Decimal;
+use serde::Deserialize;
+use serde::Serialize;
+use trade::cfd::calculate_long_liquidation_price;
+use trade::cfd::calculate_pnl;
+use trade::cfd::calculate_short_liquidation_price;
+use trade::cfd::BTCUSD_MAX_PRICE;
+use trade::Direction;
+
+/// A payout point representing a payout for a given outcome.
+#[derive(Clone, Debug, Eq, PartialEq, Serialize, Deserialize)]
+pub struct PayoutPoint {
+    /// The event outcome.
+    pub event_outcome: u64,
+    /// The payout for the outcome.
+    pub outcome_payout: u64,
+    /// Extra precision to use when computing the payout.
+    pub extra_precision: u16,
+}
+
+/// Build a [`PayoutFunction`] for an inverse perpetual future e.g. BTCUSD. Perspective is always
+/// from the person who offers, i.e. in our case from the coordinator.
+///
+/// Returns a Vec<(PayoutPoint, PayoutPoint)>>. Each tuple is meant to be put into one
+/// [`dlc_manager::payout_curve::PolynomialPayoutCurvePiece`]
+#[allow(clippy::too_many_arguments)]
+pub fn build_payout_function(
+    quantity: f32,
+    coordinator_collateral: u64,
+    trader_collateral: u64,
+    initial_price: Decimal,
+    leverage_trader: f32,
+    leverage_coordinator: f32,
+    // TODO: use the fee
+    _fee: u64,
+    coordinator_direction: Direction,
+) -> Result<Vec<(PayoutPoint, PayoutPoint)>> {
+    let mut pieces = vec![];
+    let total_collateral = coordinator_collateral + trader_collateral;
+
+    // 1. we need to calculate the lower and upper bounds. This depends on the direction of the
+    let (liquidation_price_lower_bound, liquidation_price_upper_bound) =
+        calculate_payout_curve_bounds(
+            initial_price,
+            leverage_trader,
+            leverage_coordinator,
+            coordinator_direction,
+        )?;
+
+    let (short_leverage, long_leverage) = {
+        if coordinator_direction == Direction::Long {
+            (leverage_trader, leverage_coordinator)
+        } else {
+            (leverage_coordinator, leverage_trader)
+        }
+    };
+
+    let (lower_range_lower, lower_range_upper) = calculate_lower_range_payouts(
+        coordinator_direction,
+        total_collateral,
+        liquidation_price_lower_bound,
+    )?;
```

Comment:
🔧 What about `short_liquidation_range_lower`? In ItchySats for the upper one we were using the word `threshold` too.

Reply:
it will be `(long_liquidation_range_lower, long_liquidation_range_upper`) because the lower range is when the long person gets liquidated. ,